packages/core/src/contrib/Functions.ts

```diff
@@ -1431,6 +1555,195 @@ export const compDict = {
   },
 };
 
+/*
+  Computes the signed distance for a line 
+  float sdSegment( in vec2 p, in vec2 a, in vec2 b )
+  {
+    vec2 pa = p-a, ba = b-a;
+    float h = clamp( dot(pa,ba)/dot(ba,ba), 0.0, 1.0 );
+    return length( pa - ba*h );
+  }
+*/
+const sdLine = (s: Line, p: ad.Num[]): ad.Num => {
+  return sdLineAsNums(s.start.contents, s.end.contents, p);
+};
+
+const sdLineAsNums = (a: ad.Num[], b: ad.Num[], p: ad.Num[]): ad.Num => {
+  const pa = ops.vsub(p, a);
+  const ba = ops.vsub(b, a);
+  const h = clamp([0, 1], div(ops.vdot(pa, ba), ops.vdot(ba, ba)));
+  return ops.vnorm(ops.vsub(pa, ops.vmul(h, ba)));
+};
+
+const sdPolyline = (s: Polyline, p: ad.Num[]): ad.Num => {
+  const dists: ad.Num[] = [];
+  for (let i = 0; i < s.points.contents.length - 1; i++) {
+    const start = s.points.contents[i];
+    const end = s.points.contents[i + 1];
```

Comment:
Just to make sure I understand: this works because polylines cannot be closed?

Reply:
Yes.